examples/kubernetes/cilium.yaml

```diff
@@ -26,6 +26,14 @@ data:
   debug: "false"
   disable-ipv4: "false"
 ---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cilium-metrics-config
+  namespace: kube-system
+data:
+  prometheus-serve-addr: ":9090"
```

Comment:
Is this the only change necessary to enable prometheus?

Can't the `prometheus-serve-addr` be in the cilium-config `ConfigMap`?

Reply:
I think @abhide15 is mimicking the ConfigMap in prometheus.yaml (https://github.com/cilium/cilium/blob/master/examples/kubernetes/prometheus.yaml#L9-L27)

@abhide15 If we're making it a default thing to have the cilium agent listen on the prometheus port we can simplify our configuration considerably. In particular:
- We can remove the env variable at https://github.com/cilium/cilium/blob/master/examples/kubernetes/cilium.yaml#L137-L144 and hard-code this in the command arguments
- We can remove the `cilium-metrics-config` ConfigMap completely since we won't use the data in it. If we wanted to keep port a configurable thing, we could also just fold the data into the `cilium-config` ConfigMap as aanm suggests.

Sadly, I don't think there's a way to use this variable in all the places we use the port (e.g. https://github.com/cilium/cilium/blob/master/examples/kubernetes/cilium.yaml#L93 and https://github.com/cilium/cilium/blob/master/examples/kubernetes/cilium.yaml#L112) so hardcoding might be the most consistent choice.

Reply:
@aanm We need a separate configMap named: "cilium-metrics-config". 
I like @raybejjani idea of consolidating these configMaps.

Also, the decision to start/stop metrics HTTP server is based on whether "prometheus-serve-addr" is empty or not. I would have rather have a param "enabled: true/false". 
Just makes it clear. 
@raybejjani @aanm any thoughts on this? 

I will add a separate patchset to consolidate these 2 configmaps.